silverback/application.py

```diff
@@ -72,6 +97,23 @@ def __init__(self, settings: Optional[Settings] = None):
             f"{signer_str}{start_block_str}{new_block_timeout_str}"
         )
 
+    def broker_task_decorator(
+        self,
+        task_type: TaskType,
+        container: Union[BlockContainer, ContractEvent, None] = None,
+    ):
+        def add_taskiq_task(handler: Callable):
+            # TODO: Support generic registration
+            task = self.broker.register_task(
+                handler,
+                task_name=handler.__name__,
```

Comment:
suggestion: might want to review everything that depends on `task_name` since this is changing now.  Before it was a deterministic string and now it's an arbitrary name given by the user to the func.

Reply:
Here's an audit of everything it touches as of this PR. #64 adds more, but will go over that separately.

The only thing of note we are doing with this PR is adding `task_name` to `message.labels`, as `TaskiqResult` doesn't have a `.task_name` property.
```
$ grep task_name -r silverback
silverback/application.py:                task_name=handler.__name__,
silverback/middlewares.py:    task_id = message.task_name
silverback/middlewares.py:            ",".join(f"{k}={v}" for k, v in message.labels.items() if k != "task_name")
silverback/middlewares.py:            return f"{message.task_name}[{labels_str}]"
silverback/middlewares.py:            return message.task_name
silverback/middlewares.py:        message.labels["task_name"] = message.task_name
```

Otherwise, we generate the display logs a little differently now that shows the actual user's function name (which is more useful):
```
SUCCESS: app_startup - 0.000s (0.0%)                                                       
SUCCESS: exec_event1[block=19620148,txn_id=0xef7e70a653d86386153655a84567e3fb98abf782125430c5afd1986993a77525,log_index=0] - 0.000s (0.0%)
SUCCESS: exec_block[number=19620148,hash=0x22a9f5e36eb5c18ca6642149bc6480514c92ef65b9a59ef241f6ce2cccec31cb] - 0.580s (4.5%)
SUCCESS: exec_event1[block=19620148,txn_id=0xef7e70a653d86386153655a84567e3fb98abf782125430c5afd1986993a77525,log_index=5] - 0.000s (0.0%)
...  # Error stack trace....
ERROR: exec_event1[block=19620148,txn_id=0xa4be39b9c00f2a8be111e03ca65d16259fedc0368e0a5cd5b4d6acee7f7e77bc,log_index=472] - 0.000s (0.0%)
```

I removed a lot of the context-specific logic that we were using `task_name` for before

---

If #64, I am leveraging the `task_name` label that's given to `TaskiqResult` to create the new record type for the recorder subsystem (which gets serialized to disk now):

```
silverback/recorder.py:    def _extract_custom_metrics(cls, result: Any, task_name: str) -> dict[str, Datapoint]:
silverback/recorder.py:            logger.warning(f"Cannot handle return type of '{task_name}': '{type(result)}'.")
silverback/recorder.py:                    f"Cannot handle type of metric '{task_name}.{metric_name}':"
silverback/recorder.py:        task_name = result.labels.pop("task_name", "<unknown>")
silverback/recorder.py:            task_name=task_name,
silverback/recorder.py:            metrics=cls._extract_custom_metrics(result.return_value, task_name),
```

Here is what the record stream looks like:

```jsonl
{"task_name":"app_startup","execution_time":0.0,"error":null,"completed":"2024-04-06T04:50:34.594024+00:00","block_number":null,"metrics":{"block_number":{"type":"scalar","data":0}}}
{"task_name":"exec_event1","execution_time":0.0,"error":null,"completed":"2024-04-06T04:50:37.648587+00:00","block_number":19594412,"metrics":{"amount":{"type":"scalar","data":410694555}}}
{"task_name":"exec_block","execution_time":0.36,"error":null,"completed":"2024-04-06T04:50:37.720697+00:00","block_number":19594412,"metrics":{"result":{"type":"scalar","data":149}}}
{"task_name":"exec_event1","execution_time":0.0,"error":null,"completed":"2024-04-06T04:50:37.852210+00:00","block_number":19594412,"metrics":{"amount":{"type":"scalar","data":5387740000}}}
{"task_name":"exec_event1","execution_time":0.0,"error":"Oopsie!","completed":"2024-04-06T04:50:38.054161+00:00","block_number":19594412,"metrics":{}}
```